src/runtime_src/core/include/xrt/xrt_aie.h

```diff
@@ -163,6 +163,58 @@ class bo : public xrt::bo
   }
 };
 
+class event_impl;
+class event 
+{
+public:
+
+  /**
+   * event() - Constructor from a device
+   *
+   * @param device
+   *  The device on which the profiling event should start
+   *
+   */
+  event(const xrt::device& device);
+
+  /**
+   * start_profiling() - Start AIE performance profiling
+   *
+   * @param option
+   *  Profiling option
+   * @param port1_name
+   *  Profiling port 1 name.
+   * @param port2_name
+   *  Profiling port 2 name.
+   * @param value
+   *  The number of bytes to trigger the profiling event.
+   *
+   * This function configures the performance counters in AI Engine by given
+   * port names and value. The port names and value will have different
+   * meanings on different options.
+   */
+  int 
+  start_profiling(int option, const std::string& port1_name, const std::string& port2_name, uint32_t value) const;
+
+  /**
+   * read_profiling() - Read the current performance counter value
+   *                    associated with the profiling handle
+   */
+  uint64_t
+  read_profiling() const;
```

Comment:
What does the profiling counter value represent?  I am beginning to wonder who the consumer of these event APIs will be?  Will it be end-user from application, or will it be internal XRT (xdp?) using this?

Reply:
The consumer is the end user from application. To understand the adf APIs I referred [UG-1076](https://docs.xilinx.com/r/en-US/ug1076-ai-engine-environment/Event-Profile-APIs-for-Graph-Inputs-and-Outputs)